lib/data_completeness.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+# The completeness of each type of member data as a percentage
+class DataCompleteness
+  # @param [Array PersonCard]
+  def initialize(people:)
+    @people = people
+  end
+
+  CARDS = %i(social bio contacts identifiers).freeze
+  Percentages = Struct.new(*CARDS)
+  # The percentage of data completeness for categories defined in CARDS
+  # Return [<struct DataCompleteness::Percentages>]
+  def percentages
+    Percentages.new(*CARDS.map { |card| calculate_completeness(card) })
+  end
+
+  private
+
+  attr_reader :people
+
+  def calculate_completeness(card)
+    ((people.count { |p| p.send(card.to_s).any? } / people.count.to_f) * 100).floor
+  end
```

Comment:
This is really a query method, but it's named as if it's a command method. (See, for example, http://martinfowler.com/bliki/CommandQuerySeparation.html)

Methods that are returning you something should simply be named for what they return, not what they _do_ to return that — so here just calling this `completeness` would be better.